File: bench_redis/redis.js
#!/usr/bin/env node


var client = require("redis").createClient();
var counter = 1;
var start = new Date();
 
client.on("error", function (err) {
        console.log("Error: ", err);
});


setInterval(function() {
  var delta = (new Date()) - start;
  console.log("" + Math.round(counter*1000 / delta) + " msg/s");
}, 1000);
 
client.on("ready", function () {
        var f = function (g) {
                counter++;
                client.publish("ch-1", "Message " + counter);
                // (counter % 10000 == 0) && (console.log(counter));
                g(f);
        };
 
        var g = function (f) {
                process.nextTick(function () { f(g); });
        };
 
        f(g);
});

File: bench/redis.rb
#!/usr/bin/env ruby

require 'rubygems'
require 'redis'
require 'json'

$redis = Redis.new

msg = (1..512).to_a.to_json

i = 1
t1 = Time.now
loop do
  $redis.publish "test_channel", msg
  i = i+1
  if i % 10000 == 0
    delta = Time.now - t1
    if delta > 0
      puts "#{(i / delta).round} msg/s"
    end
  end
end